phalcon/cache/backend/file.zep

```diff
@@ -629,12 +652,47 @@ class File extends Backend
 	public function getKey(key) -> string
 	{
 		if this->_useSafeKey === true {
+			if this->_useSubDir === true {
+				return  this->getSubDir(md5(key)). md5(key);
```

Comment:
space again, all over the code you should check white space. I'm also not sure exactly what this is doing but concatenating the same md5 twice looks suspect. If not at least due to the performance impact. md5 might also be over kill for splitting directories. You're just trying to bucket things aren't you? It does want an a fairly even distribution but doesn't strictly need to be cryptographically secure.

It looks like what you have is hash the key then produce: (/hash[i]).../hash for i = 0 to depth - 1?

If you want something with better performances but good distribution there are probably better things than MD5. Really you might want to investigate what hash tables tend to use as I doubt they use a cryptographically secure hash.

You might also want to consider that MD5 is 64bit. Though you're representation is actually 4 bits a byte in hexidecimal and you're taking a byte per depth. You need a max depth check of 16 in that case.

Reply:
This is an optional option, I think developers should choose according to the actual situation. Safe keys and multi-level directory storage are optional, but multi-level directory storage is built on the basis of safe keys.

I will solve the format problem as soon as possible.

Reply:
I think realistically it might be a lot clearer if extending the file class rather than modifying it. Even if it were to support nesting I think that needs to be agnostic (so an interface for directory provider for example).

If modifying a core library that's used by everyone I think that it's really worth considering the best all round results that are good for everyone or at least the most common use cases.

Everything here is a suggestion except something like let hash = mdy(key) or reuse the key var if md5 is going to be used more than once.

For the hashing this tends to be a top contender but there's no PHP extension (but there is a native php library)...

https://github.com/Cyan4973/xxHash:

Reply:
Actually there is an extension in the works: https://github.com/Megasaxon/php-xxhash